filelog.go

```diff
@@ -145,13 +154,13 @@ func NewFileLogWriter(fname string, rotate bool) *FileLogWriter {
 					(w.maxsize > 0 && w.maxsize_cursize >= w.maxsize) {
 					if err := w.handleRotate(now); err != nil {
 						fmt.Fprintf(os.Stderr, "FileLogWriter(%q): %s\n", w.filename, err)
-						return
+						rotationFailures += 1
 					}
 				} else if w.daily && now.Day() != w.daily_opendate {
 					// Since we crossed the time boundary, back the date up by one day
 					if err := w.handleRotate(now.Add(-1 * 24 * time.Hour)); err != nil {
 						fmt.Fprintf(os.Stderr, "FileLogWriter(%q): %s\n", w.filename, err)
-						return
+						rotationFailures += 1
```

Comment:
Same. In our agent `w.rot` is basically not used, it's a manual way for the agent to trigger a log roll.